src/main/java/com/google/sps/data/Utility.java

```diff
@@ -61,179 +59,39 @@ public static String graphToJson(MutableGraph<GraphNode> graph, HashSet<String>
-      default:
-        // unrecognized mutation type
-        return false;
-    }
-    return true;
-  }
-
-  /*
-   * Modify the list of tokens for graph node 'node' to accomodate the mutation
-   * 'tokenMut'. This could involve adding or removing tokens from the list.
-   *
-   * @param node the node in the graph to change the tokens of
-   * @param tokenMut the kind of mutation to perform on node of the graph
+  /**
+   * Returns a mutable shallow copy of the given guava graph
    *
-   * @return true if the change is successful, false otherwise
+   * @param graph the guava graph to shallow copy
+   * @return a shallow copy of the given guava graph
    */
-  private static boolean changeNodeToken(GraphNode node, TokenMutation tokenMut) {
-    // List of tokens to add/remove from the existing list
-    List<String> tokenNames = tokenMut.getTokenNameList();
-    // The existing list of tokens in the node
-    List<String> tokenList = node.tokenList();
-    TokenMutation.Type tokenMutType = tokenMut.getType();
-    if (tokenMutType == TokenMutation.Type.ADD_TOKEN) {
-      tokenList.addAll(tokenNames);
-    } else if (tokenMutType == TokenMutation.Type.DELETE_TOKEN) {
-      tokenList.removeAll(tokenNames);
-    } else {
-      // unrecognized mutation
-      return false;
-    }
-    return true;
+  public static MutableGraph<GraphNode> copyGraph(MutableGraph graph) {
+    return Graphs.copyOf(graph);
   }
 
   /**
-   * Alternative function for calculating maxDepth
+   * Returns a mutable deep copy of the set of strings
    *
-   * @param graphInput the input graph, as a Mutatable Graph
-   * @param roots the name (string) of the roots
-   * @param graphNodesMap a mapping of strings to GraphNodes
-   * @param maxDepth the maximum depth of a node from a root
-   * @return a graph with nodes only a certain distance from a root
+   * @param roots the set of strings to deep copy
+   * @return a deep copy of the given set of strings
    */
-  public static MutableGraph<GraphNode> getGraphWithMaxDepth(
-      MutableGraph<GraphNode> graphInput,
-      Set<String> roots,
-      HashMap<String, GraphNode> graphNodesMap,
-      int maxDepth) {
-
-    MutableGraph<GraphNode> graphToReturn = GraphBuilder.directed().build();
-    if (maxDepth < 0) {
-      return graphToReturn; // If max depth below 0, then return an emtpy graph
-    }
-
-    Map<GraphNode, Boolean> visited = new HashMap<>();
-
-    for (String rootName : roots) {
-      GraphNode rootNode = graphNodesMap.get(rootName);
-      dfsVisit(rootNode, graphInput, visited, graphToReturn, maxDepth);
-    }
-    for (EndpointPair<GraphNode> edge : graphInput.edges()) {
-      if (graphToReturn.nodes().contains(edge.nodeU())
-          && graphToReturn.nodes().contains(edge.nodeV())) {
-        graphToReturn.putEdge(edge.nodeU(), edge.nodeV());
-      }
-    }
-
-    return graphToReturn;
+  public static HashSet<String> copyRoots(HashSet<String> roots) {
```

Comment:
Why is this in utility? I think it makes more sense to have a generic copySet function here? Bc if you just copy the roots that seems specific to the DataGraph and should probably go there

Reply:
and same thing, add tests?

Reply:
I moved it back to DataServlet. Also, since I'm just using an API functions to copy the roots do you think I would still need to test?